### ISSUE TITLE
DNS resolution not working when using VPN with Split DNS

### ISSUE THREAD

**Issue by rmueller83** [netty/netty#13880]
Hi,

I am using Lettuce 6.3.1 with Netty 4.1.107 trying to connect to a Redis instance behind a OpenVPN.
Without netty-resolver-dns the connection works, but with netty-resolver-dns the connection attempt results in
`io.netty.resolver.dns.DnsErrorCauseException: Query failed with NXDOMAIN`

I am on MacOS using Viscosity with Split DNS: https://www.sparklabs.com/support/kb/article/configuring-dns-and-wins-settings/
The problem occurs with Java 8 and Java 11.

Since it is working without netty-resolver-dns it seems that there is some bug or something missing.

### Netty version
4.1.107.Final

### JVM version (e.g. `java -version`)
openjdk version "11.0.22" 2024-01-16 LTS
OpenJDK Runtime Environment Zulu11.70+15-CA (build 11.0.22+7-LTS)
OpenJDK 64-Bit Server VM Zulu11.70+15-CA (build 11.0.22+7-LTS, mixed mode)

### OS version (e.g. `uname -a`)
Darwin HOSTNAME 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:44 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6000 arm64

**Comment by normanmaurer** [comment]
@rmueller83 is the native macOS resolver used ?

**Comment by rmueller83** [comment]
Yes, you can see the dependency in my sample project and in the Log output I have
`(io.netty.util.internal.NativeLibraryLoader loadLibrary) Successfully loaded the library /var/folders/9z/xc_35vms7j312sbvw431w0300000gn/T/libnetty_resolver_dns_native_macos_aarch_646436961493426292824.dylib`

**Comment by rmueller83** [comment]
Still broken with Netty 4.1.119.Final and 4.2.0.RC4 on macOS 15.3.2. Is there anything more I can provide to debug this issue? The problem is that we don't use Netty directly but through frameworks like Lettuce and Reactor Core where I do not see the possibility to make them work without netty-resolver-dns.
